powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/v2/SecureVaultController.java

```diff
@@ -86,12 +86,13 @@ public void setSecureVaultServiceV2(SecureVaultService secureVaultServiceV2) {
         try {
             PowerAuthSignatureHttpHeaderValidator.validate(header);
         } catch (InvalidPowerAuthHttpHeaderException ex) {
-            throw new PowerAuthAuthenticationException(ex.getMessage());
+            logger.warn("Signature validation failed, error: {}", ex.getMessage());
+            throw new PowerAuthAuthenticationException("POWER_AUTH_SIGNATURE_INVALID");
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.